core/services/ping/ping360_ethernet_prober.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+import socket
+from typing import List
+
+import psutil
+from loguru import logger
+
+
+def list_ips() -> List[str]:
+    """
+    Returns a list of IPs found on ethernet interfaces that are currently up
+    """
+    stats = psutil.net_if_stats()
+
+    available_networks = []
+    ips: List[str] = []
+    for interface in stats.keys():
+        if interface in stats and getattr(stats[interface], "isup"):
+            available_networks.append(interface)
+
+    network_interface_addresses = psutil.net_if_addrs()
+    for interface in available_networks:
+        ips.extend(phys.address for phys in network_interface_addresses[interface] if phys.family == socket.AF_INET)
+    return list(set(ips))  # make sure there are not duplicated entries
```

Comment:
```suggestion
    return set(ips)  # make sure there are not duplicated entries
```
If a `set` is being used to avoid duplicates, the order is already effectively random. Is the result of this being indexed into? If not (i.e. it's just being iterated over) then it can potentially be left as a `set`.